File: app/src/main/java/com/mrebollob/loteriadenavidad/domain/entities/LotteryTicket.java
package com.mrebollob.loteriadenavidad.domain.entities;

/**
 * @author mrebollob
 */
public class LotteryTicket {

    private int id;
    private String label;
    private int number;
    private float bet;
    private float prize;
    private LotteryType lotteryType;

    public LotteryTicket() {
    }

    public LotteryTicket(String label, int number, float bet, float prize, LotteryType lotteryType) {
        this.label = label;
        this.number = number;
        this.bet = bet;
        this.prize = prize;
        this.lotteryType = lotteryType;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getLabel() {
        return label;
    }

    public void setLabel(String label) {
        this.label = label;
    }

    public int getNumber() {
        return number;
    }

    public void setNumber(int number) {
        this.number = number;
    }

    public float getBet() {
        return bet;
    }

    public void setBet(float bet) {
        this.bet = bet;
    }

    public float getPrize() {
        return prize;
    }

    public void setPrize(float prize) {
        this.prize = prize;
    }

    public LotteryType getLotteryType() {
        return lotteryType;
    }

    public void setLotteryType(LotteryType lotteryType) {
        this.lotteryType = lotteryType;
    }
}

File: app/src/main/java/com/mrebollob/loteriadenavidad/domain/entities/LotteryTicket.kt
/*
 * Copyright 2016 Manuel Rebollo Báez.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.mrebollob.loteriadenavidad.domain.entities

import com.mrebollob.loteriadenavidad.utils.PositionsFactory
import java.util.*

const val LOCAL_ID = "localId"

fun generateLocalId(): String = LOCAL_ID + "_" + UUID.randomUUID().toString().replace("-".toRegex(), "")

enum class Color {
    RED, YELLOW, GREEN, BLUE, WHITE
}

data class LotteryTicket(val localId: String = generateLocalId(),
                         val label: String = "",
                         val number: Int? = null,
                         val bet: Float? = null,
                         val prize: Float? = null,
                         val color: Color = Color.WHITE,
                         val position: Long = object : PositionsFactory {}.newPosition()) {

}
